### PR TITLE
Allow users to upload data to existing datasets

### PR DESCRIPTION
**Pull request by alesnarx**
*Issue #, if available:*
#3 

*Description of changes:*
Functionality to allow users to upload S3 data to an existing AMC data set

*Implementation:*

Step 2 - Can create a new dataset or add to an existing dataset with multiple S3 files
- Validation on unique AMC dataset names on creation

Screen Shot 2023-02-03 at 3 57 40 PM

Screen Shot 2023-02-03 at 3 59 06 PM

Step 3 - Compatibility check when uploading to an existing dataset
- Will re-use existing dataset's schema when uploading to it; if not compatible, will throw error
- If a new dataset is being created, the import and export schema options and column definitions will be enabled

Screen Shot 2023-02-03 at 3 59 39 PM

Screen Shot 2023-02-03 at 4 02 03 PM

Step 4 - Will print which columns were deleted in Step 3 under the Columns details

Screen Shot 2023-02-03 at 4 03 34 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/api/app.py [hunk 1]
```diff
@@ -65,6 +65,29 @@ def list_datasets():
...
+
+    Returns:
+
+    .. code-block:: python
+
+        {"dataSets": [...]}
+    """
+    log_request_parameters()
+    try:
+        data_set_id = app.current_request.json_body['dataSetId']
```
Comment (ikp4success) [inline source/api/app.py:81]: Can you add this new api to `test_api_integration.py`

Comment (alesnarx) [reply source/api/app.py:81]: Thanks for the comment, added in newest commit

**Comment by alesnarx** [comment]
> Tested it manually, works flawlessly. Only thing i found is the existing datasets are not organized according to recent ds created.

I also noticed this, it also seems the data sets have the same date created and date updated in the table too. Maybe we should sort by date updated. We can discuss and open a new ticket for a fix.

**Review by ikp4success — APPROVED** [review]

**Review by ikp4success — APPROVED** [review]
Approved, pending on workflow passing.

File: source/website/src/views/Step2.vue [hunk 2]
```diff
@@ -68,28 +68,126 @@ SPDX-License-Identifier: Apache-2.0
...
+                    <b-link v-b-modal.modal-encryption-mode>
+                      <b-icon-question-circle-fill variant="secondary"></b-icon-question-circle-fill>
+                    </b-link>
+                    <div class="text-muted">
+                      {{ ENCRYPTION_MODE }}
+                    </div>
+                  </b-col>
+                </b-row>
+              </div>
+              <div v-if="dataset_mode === 'JOIN'">
```
Comment (ianwow) [inline source/website/src/views/Step2.vue:178]: Put the v-if in b-row. The div is not necessary.

File: source/website/src/views/Step2.vue [hunk 3]
```diff
@@ -68,28 +68,126 @@ SPDX-License-Identifier: Apache-2.0
...
+                    <b-link v-b-modal.modal-encryption-mode>
+                      <b-icon-question-circle-fill variant="secondary"></b-icon-question-circle-fill>
+                    </b-link>
+                    <div class="text-muted">
+                      {{ ENCRYPTION_MODE }}
+                    </div>
+                  </b-col>
+                </b-row>
+              </div>
               <b-row>
```
Comment (ianwow) [inline source/website/src/views/Step2.vue:191]: The spacing between encryption mode and the country code drop down does not look right:
image

**Review by ianwow — CHANGES_REQUESTED** [review]
Try to fix the spacing issue above the country code drop down.

File: source/website/src/views/Step2.vue (hunk 3 above)
Comment (alesnarx) [reply source/website/src/views/Step2.vue:193]: I used a line break to add extra padding on the bottom of the element - let me know if you think another implementation is cleaner

Screen Shot 2023-02-21 at 5 50 12 PM

**Review by ianwow — APPROVED** [review]
great work

**Pull request merged**